tap_kafka/sync.py

```diff
@@ -182,20 +185,26 @@ def kafka_message_to_singer_record(message, primary_keys):
         "message_timestamp": get_timestamp_from_timestamp_tuple(message.timestamp()),
     }
 
-    # Add primary keys to the record message
-    for key in primary_keys:
-        pk_selector = primary_keys[key]
-        try:
-            record[key] = dpath.util.get(message.value(), pk_selector)
-        except KeyError:
-            raise PrimaryKeyNotFoundException(f"Custom primary key not found in the message: '{pk_selector}'")
+    # Add PKs to the record. In case custom PKs are defined, use them
+    if primary_keys:
+        for key in primary_keys:
+            pk_selector = primary_keys[key]
```

Comment:
Looping over a dictionary keys only to access the values using said keys is a very inefficient, use the following idiomatic way:

```suggestion
        for key, pk_selector in primary_keys.items():
```